lib/notification.js

```diff
@@ -480,15 +480,15 @@ Notification.prototype.truncateStringToLength = function (string, length, encodi
 Notification.prototype.apsPayload = function() {
 	var aps = this.payload.aps || {};
 
-	aps.badge = this.badge;
-	aps.sound = this.sound;
-	aps.alert = this.alert;
+	aps.badge = this.badge || aps.badge;
+	aps.sound = this.sound || aps.sound;
+	aps.alert = this.alert || aps.alert;
 	if (this.contentAvailable) {
 		aps["content-available"] = 1;
 	}
 	
-	aps["url-args"] = this.urlArgs;
-	aps.category = this.category;
+	aps["url-args"] = this.urlArgs || aps["url-args"];
+	aps.category = this.category || aps.category;
 
 	return Object.keys(aps).reduce(function(populated, key) {
```

Comment:
This is super duper minor, but just occurred to me: why use `reduce` over `find`? Seems like it'd be clearer?

```
return Object.keys(aps).find(function(key) {
  return aps[key] !== undefined;
}) ? aps : undefined;
```

(Technically also means it'll short-circuit as soon as it finds a populated key, but that doesn't seem like it's likely to be a meaningful performance win given the size of the objects ;-))